src/Raven.Server/Routing/RequestRouter.cs

```diff
@@ -122,6 +122,15 @@ public async ValueTask HandlePath(RequestHandlerContext reqCtx)
 
                 if (reqCtx.Database != null)
                 {
+                    if (_ravenServer.CpuCreditsAlertRaised.IsRaised())
+                    {
+                        if (_ravenServer.Configuration.Server.CpuCreditsExhaustionShouldTriggerFailover)
+                        {
+                            RejectRequestBecauseOfCpuThreshold(context);
+                            return;
+                        }
+                    }
```

Comment:
Maybe we can add an option to throttle requests then?

Reply:
I mean 'CpuCreditsExhaustionShouldTriggerFailover' is `false` then having additional option that will throttle the requests and reduce the load?

Reply:
That would just defer it, no? 
We can drop every Nth request, but that would probably suck

Reply:
If we slow down the incoming load then we should be able to recover from current state earlier, don't you think? Once we manage to recover from it then slowly decrease the throttling (not immediately)